src/App.jsx

```diff
@@ -14,33 +17,102 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const BUTTONS = {
+  ALPHABET: 'alphabet',
+  LENGTH: 'length',
+};
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+export const App = () => {
+  const [sortButton, setSortButton] = useState('');
+  const [reversed, setReversed] = useState(false);
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  function prepareList(someList) {
+    let preparedList = [...someList];
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    switch (sortButton) {
+      case BUTTONS.ALPHABET:
+        preparedList = preparedList.sort((a, b) => {
+          return a.localeCompare(b);
+        });
+        break;
+
+      case BUTTONS.LENGTH:
+        preparedList = preparedList.sort((a, b) => {
+          return a.length - b.length;
+        });
+        break;
+
+      default:
+        return preparedList;
+    }
+
+    return preparedList;
+  }
+
+  function reverseVisibleList(listToReverse, reverseParam) {
+    return reverseParam ? listToReverse.reverse() : listToReverse;
+  }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const visibleList = prepareList(goodsFromServer);
+  const acltualList = reverseVisibleList(visibleList, reversed);
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setSortButton(BUTTONS.ALPHABET)}
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortButton !== BUTTONS.ALPHABET,
+          })}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={() => setSortButton(BUTTONS.LENGTH)}
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortButton !== BUTTONS.LENGTH,
+          })}
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={() => {
+            setReversed(prev => !prev);
+          }}
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': !reversed,
+          })}
+        >
+          Reverse
+        </button>
+
+        {(sortButton || reversed) && (
+          <button
+            onClick={() => {
+              setSortButton('');
+              setReversed(false);
+            }}
```

Comment:
Move this logic from jsx to the helper reset function and use it here
```suggestion
            onClick={reset}
```